File: web/src/main/angular/app/report-body/pillbox-report/lab-report-model.ts
import {BaseReportModel} from './base-report-model';

export class LabReportModel extends BaseReportModel {
  patientId: string;
  order: string;
  date: string;
  result: string;

  constructor(data?: any) {
    super();
    if (data) Object.assign(this, data);
  }

  get patientIdCode() {
    return this.getCodeDisplay(this.patientId);
  }

  get patientIdSystem() {
    return this.getCodeSystem(this.patientId);
  }

  get dateDisplay() {
    return this.getDateDisplay(this.date);
  }
}

File: query/src/main/java/com/lantanagroup/nandina/query/pihc/fhir/r4/cerner/model/LabReportModel.java
package com.lantanagroup.nandina.query.pihc.fhir.r4.cerner.model;

import lombok.Getter;
import lombok.Setter;

@Getter @Setter
public class LabReportModel {
  private String patientId;
  private String order;
  private String date;
  private String result;
}
